### DISCUSSION TITLE
pasteing long text (500 lines) to vim is slowly

### DISCUSSION THREAD

**Discussion by r-xiumu** [wezterm/wezterm#2513] [Q&A]
When I open file by vim, paste a long text (> 500 lines). It looks like the paste process is on the screen
It is slowly. 
What should I do to improve this question？

```
vim test
:set paste
i
paste from clipboard
```

**Comment by groves** [comment]
Do you have [bracketed paste](https://cirw.in/blog/bracketed-paste) enabled in Vim? If not, it's like you're typing pastes in character by character. That page links to a plugin to turn it on.

**Comment by wez** [comment]
wezterm deliberately throttles sending large pastes and streams the content in as chunks of 1KB at a time. The reason for chunking is to avoid blocking either the application itself or wezterm if read/write buffers are filled up.
